utils.py

```diff
@@ -216,15 +216,17 @@ def db_get_many (table, data, *not_primary):
         data.append (db_decode (item))
     return data
 
-# Creates or updates an item by primary key.
+# Creates an item.
 @querylog.timed
-def db_set (table, data):
-    querylog.log_counter('db_set:' + table)
-    if db_get (table, data):
-        result = db.update_item (TableName = db_prefix + '-' + table, Key = db_encode (db_key (table, data)), AttributeUpdates = db_encode (db_key (table, data, True), True))
-    else:
-        result = db.put_item (TableName = db_prefix + '-' + table, Item = db_encode (data))
-    return result
+def db_create (table, data):
+    querylog.log_counter('db_create:' + table)
+    return db.put_item (TableName = db_prefix + '-' + table, Item = db_encode (data))
+
+# Updates an item by primary key.
+@querylog.timed
+def db_update (table, data):
+    querylog.log_counter('db_update:' + table)
+    return db.update_item (TableName = db_prefix + '-' + table, Key = db_encode (db_key (table, data)), AttributeUpdates = db_encode (db_key (table, data, True), True))
```

Comment:
> AttributeUpdates = db_encode (db_key (table, data, True), True))

Wait... isn't this supposed to be `AttributeUpdates=db_encode(data)` ?

Reply:
Maybe an APi that looks like:

```python
def db_update(table, key, updates):
  ...
```

Is clearer and safer?

Reply:
In the definition of `db_key()`, the optional 3rd argument `True` is throwing me off:

```python
def db_key (table, data, *remove):
  ...
```

This can also be called as:

```python
db_key(table, data, {})
db_key(table, data, 'asdf')
db_key(table, data, 5)
```

Wouldn't a default value for the 3rd parameter be more readable:

```python
def db_key (table, data, remove=False):
```

?